source/extensions/transport_sockets/http_11_proxy/connect.cc

```diff
@@ -37,11 +37,17 @@ UpstreamHttp11ConnectSocket::UpstreamHttp11ConnectSocket(
   // options, we want to maintain the original behavior of this transport socket.
   if (options_ && options_->http11ProxyInfo()) {
     if (transport_socket_->ssl()) {
-      header_buffer_.add(
-          absl::StrCat("CONNECT ", options_->http11ProxyInfo()->hostname, ":443 HTTP/1.1\r\n\r\n"));
+      if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.proxy_ssl_port")) {
+        header_buffer_.add(absl::StrCat(
+            "CONNECT ", options_->http11ProxyInfo()->hostname,
+            options_->http11ProxyInfo()->hostname.rfind(':') == std::string::npos ? ":443" : "",
```

Comment:
there should be utils somewhere for actual port parsing.  I'm not sure if v6 addresses are legal as proxy hosts but let's make sure we don't treat ::1 as having ports

Reply:
Done.